<?xml version="1.0"?>
<launch>
  <arg name="image_topic" default="image" />
  <arg name="init_rect_topic" default="rect" />

  <arg name="device" default="1" /> <!-- 0: cpu; 1: gpu -->
  <arg name="fp16" default="True" />

  <arg name="encoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_encoder_fp16.trt" if="$(eval device and fp16)"/>
  <arg name="decoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_decoder_fp16.trt" if="$(eval device and fp16)"/>

  <arg name="encoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_encoder.trt" if="$(eval device and not fp16)"/>
  <arg name="decoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_decoder.trt" if="$(eval device and not fp16)"/>

  <arg name="encoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_encoder.onnx" unless="$(arg device)"/>
  <arg name="decoder_model_file" value="$(find jsk_perception)/trained_data/trtr_resnet50_decoder.onnx" unless="$(arg device)"/>

  <node pkg="nodelet" type="nodelet" name="trtr_tracker"
        args="standalone jsk_perception/TrTrTrackerRos tracker_nodelet" output="screen">
    <remap from="~input" to="$(arg image_topic)" />
    <remap from="~input/rect" to="$(arg init_rect_topic)" />
    <param name="encoder_model_file" value="$(arg encoder_model_file)"/>
    <param name="decoder_model_file" value="$(arg decoder_model_file)"/>
    <param name="device" value="$(arg device)" />
  </node>

</launch>
